src/NSubstitute/Core/RobustThreadLocal.cs

```diff
@@ -20,21 +21,23 @@ public RobustThreadLocal()
 
         public RobustThreadLocal(Func<T> initialValueFactory)
         {
-            _initalValueFactory = initialValueFactory ?? throw new ArgumentNullException(nameof(initialValueFactory));
+            _initialValueFactory = initialValueFactory;
             _threadLocal = new ThreadLocal<T>(initialValueFactory);
         }
 
         public T Value
         {
             get
             {
+                // Suppress nullability for result, as we trust type by usage.
+                // For non-nullable we expect ctor with default to be used.
```

Comment:
**praise:** nice comment! 👍

Reply:
Interestingly though is that in this case nullability caught the design issue, so that it's easily possible to get null if we overlook. I would consider refactoring it in future to avoid the issue.